src/test/java/io/cryostat/net/web/http/api/v2/RuleDeleteHandlerTest.java

```diff
@@ -169,5 +176,193 @@ void shouldRespondWith404ForNonexistentRule() throws Exception {
                     Assertions.assertThrows(ApiException.class, () -> handler.handle(params));
             MatcherAssert.assertThat(ex.getStatusCode(), Matchers.equalTo(404));
         }
+
+        @Test
+        void shouldRespondWith500ForCleanupFailures() throws Exception {
+            Mockito.when(params.getPathParams()).thenReturn(Map.of("name", testRuleName));
+            MultiMap queryParams = MultiMap.caseInsensitiveMultiMap();
+            queryParams.set("clean", "true");
+            Mockito.when(params.getQueryParams()).thenReturn(queryParams);
+
+            Rule rule =
```

Comment:
What is the advantage of creating `MultiMap queryParams`, `Rule rule` and `ServiceRef serviceRef` inside each test case vs declaring them at the top of the class and using `@BeforeEach void setup()` to instantiate them?

Reply:
It's just a bit of scope cleanliness. If not every test case uses `rule` and `serviceRef` then there is no need to have them as class members and in-scope for test cases where they are unused. But, *most* of the tests within this `@Nested` class scope do use them, so maybe it would be clearer to read if those were all pulled up to class members. I'm fine with either way, I'll perform that refactor if you'd prefer it.

Reply:
I'm good with it as is